server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java

```diff
@@ -171,9 +171,12 @@ public void parse(DocumentParserContext context) throws IOException {
         }
 
         String feature = null;
+        boolean origIsWithLeafObject = context.path().isWithinLeafObject();
         try {
             // make sure that we don't expand dots in field names while parsing
-            context.path().setWithinLeafObject(true);
+            if (context.path().isWithinLeafObject() == false) {
```

Comment:
isWithinLeafObject() was not restored previously to its original value. Now we need to have that for parsing inference results.